modules/fee-market/src/lib.rs

```diff
@@ -269,35 +300,64 @@ pub mod pallet {
 				T::Currency::free_balance(&who) >= new_collateral,
 				<Error<T, I>>::InsufficientBalance
 			);
+			ensure!(
+				new_collateral > Self::relayer_locked_collateral(&who),
+				<Error<T, I>>::NewCollateralShouldLargerThanBefore
+			);
 
 			Self::update_market(
 				|| {
-					// Increase the locked collateral
-					if new_collateral >= Self::relayer_locked_collateral(&who) {
-						T::Currency::set_lock(
-							T::LockId::get(),
-							&who,
-							new_collateral,
-							WithdrawReasons::all(),
-						);
-					} else {
-						// Decrease the locked collateral
-						if let Some((_, orders_locked_collateral)) = Self::occupied(&who) {
-							ensure!(
-								new_collateral >= orders_locked_collateral,
-								<Error<T, I>>::StillHasOrdersNotConfirmed
-							);
-
-							T::Currency::remove_lock(T::LockId::get(), &who);
-							T::Currency::set_lock(
-								T::LockId::get(),
-								&who,
-								new_collateral,
-								WithdrawReasons::all(),
-							);
+					T::Currency::set_lock(
+						T::LockId::get(),
+						&who,
+						new_collateral,
+						WithdrawReasons::all(),
+					);
+
+					<RelayersMap<T, I>>::mutate(who.clone(), |relayer| {
+						if let Some(ref mut r) = relayer {
+							r.collateral = new_collateral;
 						}
+					});
+					Ok(())
+				},
+				Some(Event::<T, I>::UpdateLockedCollateral(who.clone(), new_collateral)),
+			)
+		}
+
+		/// Decrease relayer's locked collateral
+		#[pallet::weight(<T as Config<I>>::WeightInfo::decrease_locked_collateral())]
+		pub fn decrease_locked_collateral(
+			origin: OriginFor<T>,
+			new_collateral: BalanceOf<T, I>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			ensure!(Self::is_enrolled(&who), <Error<T, I>>::NotEnrolled);
+			ensure!(
+				T::Currency::free_balance(&who) >= new_collateral,
+				<Error<T, I>>::InsufficientBalance
+			);
+			ensure!(
+				new_collateral < Self::relayer_locked_collateral(&who),
+				<Error<T, I>>::NewCollateralShouldLessThanBefore
+			);
+
+			Self::update_market(
+				|| {
+					if let Some((_, orders_locked_collateral)) = Self::occupied(&who) {
+						ensure!(
+							new_collateral >= orders_locked_collateral,
+							<Error<T, I>>::StillHasOrdersNotConfirmed
+						);
 					}
 
+					T::Currency::remove_lock(T::LockId::get(), &who);
```

Comment:
The original api looks good to me, I don't think its a good ideas to change these api, since the only diff here is `remove_lock`. These api changes might require companion changes in fee-market ui and bridge, cc @JayJay1024 @fewensa .

I'm OK with the api changes if the user developers of these api think these changes are OK from them.

Reply:
I' OK. If this pr is merged, I can adapt it, and Bear also said it before. There are also two apis on the ethereum side.